Caltech-IPAC/firefly
# DM-15504: Firefly now supports using a shared and private work directories

Firefly now supports using a shared and private work directories

  - shared is for applies only to staging and upload
  - properties: `work.directory` and `shared.work.directory`
  - when `shared.work.directory` is not defined or writable then use `work.directory`
  - also includes some cleanup

**To test:**
  - make a directory to share
  - run to docker instances such as

```
docker run --rm -p 2001:8080 -e "MAX_JVM_SIZE=4G" -e "FIREFLY_SHARED_WORK_DIR=/a/b/c/shared-work-area" -v /a/b/c/shared-work-area:/a/b/c/shared-work-area --name firefly1 ipac/firefly
```
_and_
```
docker run --rm -p 2002:8080 -e "MAX_JVM_SIZE=4G" -e "FIREFLY_SHARED_WORK_DIR=/a/b/c/shared-work-area" -v /a/b/c/shared-work-area:/a/b/c/shared-work-area --name firefly2 ipac/firefly
```

Then upload a file and check to see it it ended up in the shared work area.  Do it on both instances and both files should be in the shared work area but no other work area files.  They should be in the normal work area.

_note:_
  - this test could also be done in k8s
